x-pack/platform/plugins/shared/streams/server/lib/rules/esql/lib/build_esql_search_request.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { estypes } from '@elastic/elasticsearch';
+
+export const buildEsqlSearchRequest = ({
+  query,
+  timestampField,
+  from,
+  to,
+  previousOriginalDocumentIds,
+}: {
+  query: string;
+  timestampField: string;
+  from: string;
+  to: string;
+  previousOriginalDocumentIds: string[];
+}) => {
+  const sanitizedDocumentIds = previousOriginalDocumentIds.filter((id) => id != null);
+  const rangeFilter = {
+    range: {
+      [timestampField]: {
+        lte: to,
+        gte: from,
+        format: 'strict_date_optional_time',
+      },
+    },
+  };
+
+  const requestFilter: estypes.QueryDslQueryContainer[] = [rangeFilter];
+
+  return {
+    // Make sure query includes `METADATA _id, _source`
```

Comment:
Since we have _source (or should have, seems like it's up the user but we want to fix this, is this a follow-up?), do we even need the `row_to_document` logic? Or could we simply take the source which will contain all fields and more anyway?

If we don't do this and do have `METADATA _source`, then I guess we will duplicate a lot of information (e.g. if the user does an `EVAL` in the rule, then do we need to capture that in the alert doc, since it's just derived data?

Reply:
> When/if we support full ESQL query, then I guess it will be different

Yeah, let that be the problem of future Kevin and future Joe.

I think for now just taking the source and id simplifies things